nevergrad/functions/control/test_mujoco.py

```diff
@@ -3,24 +3,49 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+import pytest
 import numpy as np
 from .core import BaseFunction
 
+
 class MountainCarContinuous(BaseFunction):
     env_name = "MountainCarContinuous-v0"
-    policy_dim= (1, 2)
+    policy_dim = (2, 1)
     state_mean = [0, 0]
     state_std = [1, 1]
 
+
 def test_gym() -> None:
-    func = MountainCarContinuous(num_rollouts=5)
+    for intermediate_layer_dim in [None, (10,), (5, 5)]:
+        for noise_level in [0.0, 0.9]:
+            for deterministic_sim in [True, False]:
+                for states_normalization in [True, False]:
+                    func = MountainCarContinuous(num_rollouts=5, intermediate_layer_dim=intermediate_layer_dim,
+                                                 noise_level=noise_level, deterministic_sim=deterministic_sim,
+                                                 states_normalization=states_normalization, random_state=42)
+
+                    x = func.parametrization.value
+                    value = func(x)
+                    np.testing.assert_almost_equal(value, 0.)
+                    x = func.parametrization.sample().value
+                    assert func(x) != 0
+                    if noise_level == 0.0 and deterministic_sim:
+                        assert func(x) == func(x)
+                    else:
+                        assert func(x) != func(x)
+
 
-    x = func.parametrization.value
-    np.testing.assert_array_equal(x, [[0., 0.]])
-    value = func(x)
-    np.testing.assert_almost_equal(value, 0.)
+def test_all_mujoco_envs() -> None:
+    pytest.importorskip('mujoco_py')
+    core = pytest.importorskip('nevergrad.functions.control.core')
 
-    x = func.parametrization.sample().value
-    assert not np.array_equal(x, [[0., 0.]])
-    value = func(func.parametrization.sample().value)
-    assert value != 0
+    for module in ["Ant", "Swimmer", "HalfCheetah", "Hopper", "Walker2d", "Humanoid"]:
+        for intermediate_layer_dim in [None, (50,)]:
+            for noise_level in [0.0, 0.9]:
+                for deterministic_sim in [True, False]:
+                    for states_normalization in [True, False]:
```

Comment:
same as above, however since this test will only run for mujoco users which will be interested in making sure all the combinations work, you can keep all combinations.
The simplest way would be to stack multiple pytest.mark.parametrize in this way:
```python
@pytest.mark.parametrize("x", [0, 1])
@pytest.mark.parametrize("y", [2, 3])
def test_foo(x, y):
    pass
```